src/test/java/com/teragrep/cfe_39/HdfsTest.java

```diff
@@ -0,0 +1,260 @@
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+
+public class HdfsTest {
+    private static final Logger LOGGER = LoggerFactory.getLogger(HdfsTest.class);
+
+    private static MiniDFSCluster hdfsCluster;
+    private static File baseDir;
+    private static Config config;
+
+    // Generate AVRO-files for testing the HDFS writes.
+    // @BeforeAll
+    public static void generateTestData() throws IOException, InterruptedException {
+        config = null;
+        try {
+            config = new Config();
+        } catch (IOException e){
+            LOGGER.error("Can't load config: " + e);
+            System.exit(1);
+        } catch (IllegalArgumentException e) {
+            LOGGER.error("Got invalid config: " + e);
+            System.exit(1);
+        }
+        startMiniCluster();
+        config.setMaximumFileSize(3000); // 10 loops (140 records) are in use at the moment, and that is sized at 36,102 bytes.
+        KafkaController kafkaController = new KafkaController(config);
+        kafkaController.run();
+    }
+
+    public static void startMiniCluster() throws IOException {
+        // Create a HDFS miniCluster
+        baseDir = Files.createTempDirectory("test_hdfs").toFile().getAbsoluteFile();
+        Configuration conf = new Configuration();
+        conf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR, baseDir.getAbsolutePath());
+        MiniDFSCluster.Builder builder = new MiniDFSCluster.Builder(conf);
+        hdfsCluster = builder.build();
+        String hdfsURI = "hdfs://localhost:"+ hdfsCluster.getNameNodePort() + "/";
+        LOGGER.debug("hdfsURI: " + hdfsURI);
+        config.setHdfsuri(hdfsURI);
+        DistributedFileSystem fileSystem = hdfsCluster.getFileSystem();
+    }
+
+    // Delete the generated AVRO-files.
+    // @AfterAll
+    public static void deleteTestData() throws IOException {
+        Path queueDirectory = new Path(config.getQueueDirectory()); // Paths.get(config.getQueueDirectory());
+        for (int j = 0; j <= 9; j++) {
+            for (int i = 1; i <= 2; i++) {
+                File syslogFile = new File(
+                        queueDirectory.toUri()
+                                + File.separator
+                                + "testConsumerTopic"
+                                + j
+                                + "."
+                                + i
+                );
+                try {
+                    boolean result = Files.deleteIfExists(syslogFile.toPath()); //surround it in try catch block
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+        closeMiniCluster();
+    }
+
+    public static void closeMiniCluster() {
+        // Teardown HDFS miniCluster
+        hdfsCluster.shutdown();
+        FileUtil.fullyDelete(baseDir);
+    }
+
+    // @Test
+    public void miniClusterDebugging() throws InterruptedException, IOException {
```

Comment:
Tests should not throw nor try/catch anything, use `Assertions.assertDoesNotThrow`, `Assertions.assertThrows`, `Assertions.assertAll` and such instead